src/geointerface.jl

```diff
@@ -0,0 +1,98 @@
+using GeoInterface
+
+const GI = GeoInterface
+
+# Coord is a PointTrait
+GI.isgeometry(::Type{Coord}) = true
+GI.geomtrait(::Coord) = GI.PointTrait()
+GI.x(::PointTrait, c::Coord) = c.x
+GI.y(::PointTrait, c::Coord) = c.y
+GI.z(::PointTrait, c::Coord) = c.z
+GI.getcoord(::PointTrait, c::Coord) = (c.x, c.y, c.z)
+GI.getcoord(::PointTrait, c::Coord, i::Int) = c[i]
+coordnames(::PointTrait, ::Coord) = (:X, :Y, :Z)
+
+"""
+    reproject(geometry, source_crs, target_crs)
+
+Reproject any GeoInterface.jl compatible `geometry` from `source_crs` to `target_crs`.
+
+The returned object will be constructed from `GeoInterface.WrapperGeometry`
+geometries, wrapping views of `Proj.Coord`.
+"""
+function reproject(geom, source_crs, target_crs; time=Inf)
```

Comment:
Can we add this as a method stub to GeoInterface so that other packages (ArchGDAL?) can specialize on this?

Reply:
Does that make sense though? It wouldn't really give us anything more than we have.

The point of this PR is that it converts *any* geointerface object using Proj. There is no way to dispatch on that, so a stub would necessitate type piracy. I already wrote a `ArchGDAL.reproject` method years ago. It will also handle any GeoInterface geometries when this is merged:
https://github.com/yeesian/ArchGDAL.jl/pull/366

But you have to use `ArchGDAL.reproject` for that to work. If we use a stub method in GeoInterface.jl then the packages can only handle types they own, so `ArchGDAL.AbstractGeometry` for ArchGDAL.jl, and nothing for Proj.jl.

I'm trying to shift everything to package methods because the methods in GeoInterface (like `area` etc) don't really make sense, there is no way to add useful methods to them without piracy or a backend system. Mostly I use Shapefile.jl and GeoJSON.jl directly, and need a way to work with those objects.

Reply:
Hmm, I guess I can see that.  I guess packages could implement functions which take arbitrary GeoInterface geometries and return their own internal structs then?

Reply:
Yep. That's what my PRs in [LibGEOS](https://github.com/JuliaGeo/LibGEOS.jl/pull/156), [ArchGDAL](https://github.com/yeesian/ArchGDAL.jl/pull/366) and this one are doing.

I'm trying to make them at least be the same same and syntax in all packages. 

For example `reproject` should be the essentially the same in ArchGDAL, Rasters and Proj. (Rasters will need some changes)